File: lab4/lang-tests/test-8/passage_of_parameters.pl
$x=50;
$y=70;

print "\n";
print "BEFORE subrutine.\n";
print "x --> $x\n";
print "y --> $y\n";
print "\n";

print "START subrutine\n";
&interchange ($x, $y);
print "END subrutine\n";
print"\n";

print "AFTER subrutine.\n";
print "x:$x y:$y\n";

print "\nEl programa realiza pasaje por referencia(call_by_reference)\n";
print "el valor de las variables sufre modificacion fuera de la subrutina\n\n";

sub interchange {
my($x1, $y1) = @_;
print "--parameters--\n";
print "\t x1: $x1\n\t y1: $y1\n";
@_[0] = $y1;
@_[1] = $x1; 
#local $z1;
#$z1 = $x1;
#$x1 = $y1;
#$y1 = $z1;

print "after operations in subrutine\n";
print "\t x1:$x1\n\t y1:$y1\n";
}

#esto hace la profe, tiene que ser pasaje por referencia
#my($a, $b) = @_;
#@_[0] = $b;
#@_[1] = $a;

File: lab4/lang-tests/test-8/passage_of_parameters.c
#include <stdio.h>

void interchange(int x1, int y1){
    
    int z1 = 0;
    printf("\nInside interchange function before interchange\n");
    printf("x1 = %d \ny1 = %d\n", x1, y1);

    z1 = x1;
    x1 = y1;
    y1 = z1;

    printf("Inside interchange function after interchange\n");
    printf("x1 = %d\ny1 = %d\n", x1, y1);

}

int main(void){
    
    int x=50, y=70;
    printf("\nfunction main:\n");
    printf("x = %d\ny = %d\n", x, y);

    interchange(x,y);
    printf("\nfunction main:\n");
    printf("x = %d\ny = %d\n", x, y);
    printf("\nEl programa realiza pasaje por valor (call_by_value)\n\n");

    return 0;

}